wandb/sdk/wandb_config.py

```diff
@@ -90,6 +94,7 @@ class Config(object):
 
     def __init__(self):
         object.__setattr__(self, "_items", dict())
+        object.__setattr__(self, "_artifact_items_map", dict())
```

Comment:
I introduce this second mapping, so users can access their artifacts in the run.config directly from their desired name without having to go through the artifacts key.

Eg. 
```
model = run.use_artifact("name", ds_slot_name="dataset")
```
```
model = run.use_artifact("name")
run.config.dataset = model
```
Only have the artifact in the config once, under the dataset config slot.